loki/tools/util.py

```diff
@@ -794,3 +794,21 @@ def graphviz_present():
         return False
 
     return True
+
+
+def resolve_typebound_var(name, variable_map):
+    """
+    A small convenience utility to resolve type-bound variables.
+
+    Parameters
+    ----------
+    name : str
+        The full name of the variable to be resolved, e.g., a%b%c%d.
+    variable_map : dict
+        A map of the variables defined in the current scope.
+    """
+
+    name_parts = name.split('%', maxsplit=1)
+    if (var := variable_map.get(name_parts[0], None)) and len(name_parts) > 1:
+        var = var.get_derived_type_member(name_parts[1])
+    return var
```

Comment:
Ah yes, that would be much cleaner, thanks! I assume you mean `ProgramUnit` though rather than `Scope`, because that is where the `variables` and `variable_map` are computed?

Reply:
Yes, thanks, `ProgramUnit` would be the right place.